internal/install/command.go

```diff
@@ -84,39 +94,43 @@ var Command = &cobra.Command{
 	},
 }
 
-func assertProfileIsValid(maxTimeoutSeconds int) error {
+func assertProfileIsValid(maxTimeoutSeconds int, sg *segment.Segment) error {
+
 	accountID := configAPI.GetActiveProfileAccountID()
+	sg.Track(accountID, segment.NewEvent(segment.EventTypes.InstallStarted, ""))
+
 	if accountID == 0 {
+		sg.Track(accountID, segment.NewEvent(segment.EventTypes.AccountIDMissing, ""))
 		return fmt.Errorf("accountID is required")
 	}
 
 	if configAPI.GetActiveProfileString(config.APIKey) == "" {
+		sg.Track(accountID, segment.NewEvent(segment.EventTypes.APIKeyMissing, ""))
 		return fmt.Errorf("API key is required")
 	}
 
 	if configAPI.GetActiveProfileString(config.Region) == "" {
+		sg.Track(accountID, segment.NewEvent(segment.EventTypes.RegionMissing, ""))
 		return fmt.Errorf("region is required")
 	}
 
+	if err := checkNetwork(client.NRClient); err != nil {
+		sg.Track(accountID, segment.NewEvent(segment.EventTypes.UnableToConnect, err.Error()))
+		return err
+	}
+
 	licenseKey, err := client.FetchLicenseKey(accountID, config.FlagProfileName, &maxTimeoutSeconds)
 	if err != nil {
-		return fmt.Errorf("could not fetch license key for account %d: %s", accountID, err)
+		sg.Track(accountID, segment.NewEvent(segment.EventTypes.UnableToFetchLicenseKey, err.Error()))
+		return fmt.Errorf("could not fetch license key for account %d:, license key: %v %s", accountID, utils.Obfuscate(licenseKey), err)
 	}
+	sg.Track(accountID, segment.NewEvent(segment.EventTypes.AbleToFetchLicenseKey, ""))
```

Comment:
@NRhzhao it might be good to use a different syntax for that event type so it doesn't cover the syntax for the Unable event. Maybe "LicenseKeyFetchedOk"